pySDC/implementations/convergence_controller_classes/interpolate_between_restarts.py

```diff
@@ -0,0 +1,102 @@
+import numpy as np
+from pySDC.core.ConvergenceController import ConvergenceController, Status
+from pySDC.core.Lagrange import LagrangeApproximation
+from pySDC.core.Collocation import CollBase
+
+
+class InterpolateBetweenRestarts(ConvergenceController):
+    """
+    Interpolate the solution and right hand side to the new set of collocation nodes after a restart.
+    The idea is that when you adjust the step size between restarts, you already know what the new quadrature method
+    is going to be and possibly interpolating the current iterate to these results in a better initial guess than
+    spreading the initial conditions or whatever you usually like to do.
+    """
+
+    def setup(self, controller, params, description, **kwargs):
+        """
+        Store the initial guess used in the sweeper when no restart has happened
+
+        Args:
+            controller (pySDC.Controller.controller): The controller
+            params (dict): Parameters for the convergence controller
+            description (dict): The description object used to instantiate the controller
+
+        Returns:
+            None
+        """
+        defaults = {
+            'control_order': 50,
+        }
+        return {**defaults, **super().setup(controller, params, description, **kwargs)}
+
+    def setup_status_variables(self, controller, **kwargs):
+        """
+        Add variables to the sweeper containing the interpolated solution and right hand side.
+
+        Args:
+            controller (pySDC.Controller.controller): The controller
+
+        Returns:
+            None
+        """
+        self.status = Status(['u_inter', 'f_inter', 'perform_interpolation'])
+
+        self.status.u_inter = []
+        self.status.f_inter = []
+        self.status.perform_interpolation = False
+
+    def post_spread_processing(self, controller, S, **kwargs):
```

Comment:
Could you rename `S` to `step` so it is more explicit ? And follow the new contribution guidelines :stuck_out_tongue: ...